### DISCUSSION TITLE
Z80 instruction ld hl,nn - How to get nn to reference label

### DISCUSSION THREAD

**Discussion by RobBrooks** [NationalSecurityAgency/ghidra#6201] [Debugger]
Apologies for clumsy vernacular 
The hl register is getting loaded with an immediate value which obviously represents an effective address which I currently have referenced by a label as you can see from the 'ld a' instruction a few lines below it.
Is there a command I can make use of which will switch the context of the disassembly for this instruction to make its symbol readable?
I'm guessing this must be possible because if I was to dump out a listing of this and change the code in any way, there's a good chance that this will potentially result in a misalignment which will cause the incorrect data being referenced.
Many thanks.
<img width="432" alt="Screenshot 2024-02-06 at 18 23 44" src="https://github.com/NationalSecurityAgency/ghidra/assets/17879728/b63f3db0-9912-412f-82ff-42f8a89016cf">

**Comment by philpem** [comment]
Select the address literal in the `LD` instruction's operands, then hit Ctrl-Alt-R (or right-click it, References -> Create Memory Reference).

**Comment by RobBrooks** [comment]
Fantastic. That's really helpful. Thank you very much.

Just one other related question if I may.....
If an address falls outside of the current context and instead in an overlay bank that I have defined, how can I get the reference to create a label inside a specificed bank?

For example - the address 0x8950 is being loaded into DE here in mt 'ProgramROM' section....

<img width="432" alt="Screenshot 2024-02-08 at 17 29 57" src="https://github.com/NationalSecurityAgency/ghidra/assets/17879728/cc9c1835-3a89-4399-bb41-b15f01d58a1b">

....but the data it needs to reference is in 'DataBank0'

<img width="1091" alt="Screenshot 2024-02-08 at 17 32 51" src="https://github.com/NationalSecurityAgency/ghidra/assets/17879728/eb5af3f6-9242-4f75-b0e0-d2cedb3bd074">

How do I create the reference so it appears in DataBank0 when I go and interrogate that section?

Any many thanks for your help. I've only just started using Ghidra and I'm making great headway but there are just a few things I find that are slowing me down a little and this one is pretty significant.

All the best.

**Comment by RobBrooks** [comment]
OK, I seem to have figured this out and the answer wasn't very far from the last one:

<img width="571" alt="Screenshot 2024-02-09 at 18 44 58" src="https://github.com/NationalSecurityAgency/ghidra/assets/17879728/0dc79464-7f89-45da-8677-07f2735238af">
<img width="502" alt="Screenshot 2024-02-09 at 18 45 33" src="https://github.com/NationalSecurityAgency/ghidra/assets/17879728/b04004db-3ad0-4ae5-aeb5-a0dc34a6284e">
